src/poetry/utils/password_manager.py

```diff
@@ -47,8 +47,18 @@ def get_credential(
 
         import keyring
 
+        from keyring.errors import KeyringError
+        from keyring.errors import KeyringLocked
+
         for name in names:
-            credential = keyring.get_credential(name, username)
+            credential = None
+            try:
+                credential = keyring.get_credential(name, username)
+            except KeyringLocked:
```

Comment:
is `KeyringLocked` really worth a special case?  every other access of keyring in this file goes
```python
except (RuntimeError, keyring.errors.KeyringError):
```
and I feel it would be hard to argue with consistency,

(Personally I am am not a fan of catching `RuntimeError` and would prefer to have people raise bugs against keyring if it raises random exceptions; but that isn't the view taken in the rest of the code)

Reply:
KeyLocked is an expected exception resulting from specific user interaction. That's why we wouldn't want to print a traceback. Traceback are for unexpected errors that might need further action by the user. That's the case for all other exceptions thrown by keyring.

Reply:
sure, I wouldn't print a traceback for any of them

Reply:
Well, for RuntimeError a traceback could be relevant for debugging. I think the same applies to KeyringError.

Reply:
well, I don't care enough to argue about it, happy to leave that to someone with the commit bit.  (but you'll have fewer testcases to write if you handle all the exceptions in the same way!)

Reply:
IMO, it makes sense to print the traceback (or at least the original error) for unexpected error messages. So I think that's ok with me.

As already mentioned, we should add tests.